plane/src/drone/mod.rs

```diff
@@ -178,11 +178,19 @@ impl Drone {
                     "Only one of `docker_config` and `executor_config` may be provided."
                 ));
             }
-            (Some(docker_config), None) => {
+            (Some(mut docker_config), None) => {
                 tracing::warn!("`docker_config` is deprecated. Use `executor_config` instead.");
+                docker_config.auto_prune = docker_config.auto_prune.or(config.auto_prune);
+                docker_config.cleanup_min_age =
+                    docker_config.cleanup_min_age.or(config.cleanup_min_age);
+
                 PlaneDocker::new(Docker::connect_with_local_defaults()?, docker_config).await?
             }
-            (None, Some(ExecutorConfig::Docker(docker_config))) => {
+            (None, Some(ExecutorConfig::Docker(mut docker_config))) => {
+                docker_config.auto_prune = docker_config.auto_prune.or(config.auto_prune);
+                docker_config.cleanup_min_age =
+                    docker_config.cleanup_min_age.or(config.cleanup_min_age);
```

Comment:
this is a little messy, but will be gone once we remove the now-deprecated `auto_prune` and `cleanup_min_age` from the global drone config.